library/spdm_responder_lib/libspdm_rsp_algorithms.c

```diff
@@ -463,9 +463,14 @@ libspdm_return_t libspdm_get_response_algorithms(libspdm_context_t *spdm_context
         for (index = 0; index < spdm_request->header.param1; index++) {
             switch (struct_table->alg_type) {
             case SPDM_NEGOTIATE_ALGORITHMS_STRUCT_TABLE_ALG_TYPE_DHE:
+                if (!struct_table->alg_supported) {
+                    return libspdm_generate_error_response(
+                        spdm_context, SPDM_ERROR_CODE_INVALID_REQUEST,
```

Comment:
Please add a unit test that triggers this error condition.

Reply:
Added a test to check this